lab_4_retrieval_w_clustering/main.py

```diff
@@ -26,6 +38,10 @@ def get_paragraphs(text: str) -> list[str]:
     Returns:
         list[str]: Paragraphs from document.
     """
+    if not (isinstance(text, str) and len(text) > 0):
+        raise ValueError
+    split_text = text.split("\n")
+    return split_text[:-1] if split_text[-1] == "" else split_text
```

Comment:
is it really need?

Reply:
Yes, because if the text ends with an empty line, the tokenizer would create a list with an empty string in the end. This empty string cannot be used in some functions, as they have to return None or raise errors when given an empty string.

Reply:
try just splitting by `\n`